app/scripts/modules/core/src/pipeline/config/stages/managed/ImportDeliveryConfigExecutionDetails.tsx

```diff
@@ -2,10 +2,58 @@ import React from 'react';
 
 import { ExecutionDetailsSection, IExecutionDetailsSectionProps, StageFailureMessage } from 'core/pipeline';
 import { IGitTrigger } from 'core/domain';
+import { CollapsibleSection, Markdown } from 'core/presentation';
 import { SETTINGS } from 'core/config';
 
 const NOT_FOUND = 'Not found';
 
+interface IDeliveryConfigImportErrorDetails {
+  error: string;
+  message: string;
+  pathExpression: string;
+}
+
+interface IDeliveryConfigImportError {
+  message: string;
+  details?: IDeliveryConfigImportErrorDetails;
+}
+
+const CustomErrorMessage = (message: string, debugDetails?: string) => {
+  return (
+    <div>
+      <div className="alert alert-danger">
+        <b>There was an error parsing your delivery config file.</b>
+        <br />
+        <Markdown message={message} />
+        <br />
+        <CollapsibleSection heading={({ chevron }) => <span>{chevron} Debug Details</span>}>
+          <pre>{debugDetails}</pre>
+        </CollapsibleSection>
+      </div>
+    </div>
+  );
+};
+
+function buildCustomErrorMessage(error: IDeliveryConfigImportError) {
+  if (!error) {
+    return null;
+  }
+
+  if (!error.details) {
+    return CustomErrorMessage(error.message);
+  }
+
+  const errorMessage =
+    {
+      missing_property: `The following property is missing: \`${error.details.pathExpression}\``,
+      invalid_type: `The type of property \`${error.details.pathExpression}\` is invalid.`,
+      invalid_format: `The format of property \`${error.details.pathExpression}\` is invalid.`,
+      invalid_value: `The value of property \`${error.details.pathExpression}\` is invalid.`,
```

Comment:
I have the keys as strings (e.g. `'missing_property'`) but when I commit the file it looks like there's a pre-commit hook that runs and changes that into symbols, except the linter complains they have to be camel-case. 😝 

Not sure how to fix that...